src/Auth/AuthCookie.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Authentication Cookie.
+ *
+ * @package WPGraphQL\Login\Auth
+ * @since 0.0.1
+ */
+
+ // phpcs:disable WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+
+declare(strict_types=1);
+
+namespace WPGraphQL\Login\Auth;
+
+/**
+ * Class AuthCookie
+ *
+ * Handles custom authentication cookies, including setting the SameSite attribute.
+ */
+class AuthCookie {
+	/**
+	 * Sets the authentication cookies based on user ID.
+	 * Provides an alternative to `wp_set_auth_cookie` that supports the SameSite cookie attribute.
+	 *
+	 * @param int  $user_id  User ID.
+	 * @param bool $remember Whether to remember the user.
+	 */
+	public static function wp_custom_auth_cookie( int $user_id, bool $remember = false ): void {
```

Comment:
It's namespaced to the class, we don't need to prefix the method name too.

```suggestion
	public static function set_auth_cookie( int $user_id, bool $remember = false ): void {
```